py/webdriver/edge/webdriver.py

```diff
@@ -0,0 +1,203 @@
+                    desired_capabilities=desired_capabilities)
+            else:
+                RemoteWebDriver.__init__(
+                    self,
+                    command_executor=RemoteConnection(
+                        remote_server_addr='http://localhost:%d' % self.port,
+                        resolve_ip=False,
+                        keep_alive=keep_alive),
+                    desired_capabilities=capabilities)
+        except Exception:
+            self.quit()
+            raise
+        self._is_remote = False
+
+    def launch_app(self, id):
+        """Launches Edge app specified by id."""
+        return self.execute("launchApp", {'id': id})
+
+    def get_network_conditions(self):
+        """
+        Gets Edge network emulation settings.
+
+        :Returns:
+            A dict. For example:
+
+            {'latency': 4, 'download_throughput': 2, 'upload_throughput': 2,
+            'offline': False}
+
+        """
+        return self.execute("getNetworkConditions")['value']
+
+    def set_network_conditions(self, **network_conditions):
+        """
+        Sets Edge network emulation settings.
+
+        :Args:
+         - network_conditions: A dict with conditions specification.
+
+        :Usage:
+            driver.set_network_conditions(
+                offline=False,
+                latency=5,  # additional latency (ms)
+                download_throughput=500 * 1024,  # maximal throughput
+                upload_throughput=500 * 1024)  # maximal throughput
+
+            Note: 'throughput' can be used to set both (for download and upload).
+        """
+        self.execute("setNetworkConditions", {
+            'network_conditions': network_conditions
+        })
+
+    def execute_cdp_cmd(self, cmd, cmd_args):
+        """
+        Execute Edge Devtools Protocol command and get returned result
+
+        The command and command args should follow Edge devtools protocol domains/commands, refer to link
+        https://chromedevtools.github.io/devtools-protocol/
+
+        :Args:
+         - cmd: A str, command name
+         - cmd_args: A dict, command args. empty dict {} if there is no command args
+
+        :Usage:
+            driver.execute_cdp_cmd('Network.getResponseBody', {'requestId': requestId})
+
+        :Returns:
+            A dict, empty dict {} if there is no result to return.
+            For example to getResponseBody:
+
+            {'base64Encoded': False, 'body': 'response body string'}
+
+        """
+        return self.execute("executeCdpCommand", {'cmd': cmd, 'params': cmd_args})['value']
+
+    def quit(self):
+        """
+        Closes the browser and shuts down the EdgeDriver executable
+        that is started when starting the EdgeDriver
+        """
+        try:
+            RemoteWebDriver.quit(self)
+        except Exception:
+            # We don't care about the message because something probably has gone wrong
+            pass
+        if self.use_chromium:
```

Comment:
This can just say self.service.stop() once edge_service is removed. Also, looking at the official chrome module, there should be a "finally:" here.